File: store_server/src/main/java/org/project/store_server/service/ProductService.java
package org.project.store_server.service;

import org.project.store_server.model.dto.product.ProductResponseModel;
import reactor.core.publisher.Mono;

import java.util.List;

public interface ProductService {
    Boolean checkProductAvailability(String sku);
    List<ProductResponseModel> getAllProducts();
}

File: store-frontend/src/app/services/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Product } from '../models/product.model';

@Injectable({
  providedIn: 'root'
})
export class ProductService {
  private apiUrl = 'http://localhost:8080/stocks';

  constructor(private http: HttpClient) { }

  getAllProducts(): Observable<Product[]> {
    return this.http.get<Product[]>(`${this.apiUrl}/products`);
  }

  checkProductAvailability(sku: string): Observable<boolean> {
    return this.http.get<boolean>(`${this.apiUrl}/products/${sku}/availability`);
  }
}